dhcpv6/async.go

```diff
@@ -0,0 +1,214 @@
+package dhcpv6
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+	"time"
+)
+
+// ClientAsync implements an asynchronous DHCPv6 client
+type ClientAsync struct {
+	ReadTimeout  time.Duration
+	WriteTimeout time.Duration
+	LocalAddr    net.Addr
+	RemoteAddr   net.Addr
+
+	connection   *net.UDPConn
+	cancel       context.CancelFunc
+	stopping     chan bool
+	receiveQueue chan DHCPv6
+	sendQueue    chan DHCPv6
+	packetsLock  sync.Mutex
+	packets      map[uint32](chan Response)
+}
+
+// NewClientAsync creates an asynchronous client
+func NewClientAsync() *ClientAsync {
+	return &ClientAsync{
+		ReadTimeout:  DefaultReadTimeout,
+		WriteTimeout: DefaultWriteTimeout,
+	}
+}
+
+// OpenForInterface starts the client on the specified interface
```

Comment:
mention that it requires a link-local address to work, and that it overrides the listening port to 546